editoast/src/models/timetable.rs

```diff
@@ -55,27 +56,64 @@ impl Timetable {
             .map_err(Into::into)
     }
 
-    pub async fn schedules_before_date(
+    /// This function will return all train schedules in a timetable that runs within the time window.
+    ///
+    /// **IMPORTANT**: The filter is based on the scheduled arrival time and not the actual simulated arrival time.
+    ///
+    /// The diagram below shows a list of trains in a timetable:
+    /// `?`: unscheduled arrival times.
+    /// `|`: scheduled arrival times.
+    ///
+    /// ```
+    ///                       min_time               max_time
+    ///     Time Window           |----------------------|
+    /// Train 1 ✅         |-------------|
+    /// Train 2 ❌    |-------|
+    /// Train 3 ✅                              |----------|
+    /// Train 4 ✅   |-------?
+    /// Train 5 ✅                        |---------?
+    /// Train 6 ❌                                           |------?
+    /// ```
+    pub async fn schedules_in_time_window(
         self,
         conn: &mut DbConnection,
-        time: DateTime<Utc>,
+        min_time: DateTime<Utc>,
+        max_time: DateTime<Utc>,
     ) -> Result<Vec<TrainSchedule>> {
-        use diesel::prelude::*;
-        use diesel_async::RunQueryDsl;
-        use editoast_models::tables::train_schedule::dsl;
-
-        let train_schedules = dsl::train_schedule
-            .filter(dsl::start_time.le(time))
-            .filter(dsl::timetable_id.eq(self.id))
-            .load_stream::<Row<TrainSchedule>>(conn.write().await.deref_mut())
-            .await?
-            .map_ok(|ts| ts.into())
-            .try_collect::<Vec<TrainSchedule>>()
-            .await;
-        match train_schedules {
-            Ok(train_schedules) => Ok(train_schedules),
-            Err(err) => Err(err.into()),
-        }
+        let train_schedules = sql_query("
+        WITH last_path_step AS (
```

Comment:
For big SQL queries, we try to put them in external files (helps also with syntax highlighting in our editors). See [here](https://github.com/OpenRailAssociation/osrd/blob/14fa21bf67cf757bd42881dbf5459adab5281a8c/editoast/src/models/infra/voltage.rs#L27) for example.

Reply:
Done